src/atc/etl/orchestrator.py

```diff
@@ -24,7 +25,38 @@ def step(self, etl: EtlBase) -> "Orchestrator":
     transform_with = step
     load_into = step
 
-    def execute(self) -> None:
-        datasets: dataset_group = {}
-        for step in self.steps:
+    def etl(self, inputs: dataset_group = None) -> dataset_group:
+        inputs = inputs or {}
+
+        # make a shallow copy of the inputs for waring after the first step
+        datasets = inputs.copy()
+        if not self.steps:
+            raise NotImplementedError("The orchestrator has no steps.")
+
+        # treat the fist step differently to warn in case the input was not handled
+        datasets = self.steps[0].etl(datasets)
+        if len(inputs) and (len(inputs) + 1 == len(datasets)):
```

Comment:
I do not understand this need to check if the fist step have cleaned up the inputs.
I could imagine a valid scenario where i have an input dataframe, a first step extractor reading a dataframe, and a transformer working on both of them?

Reply:
Absolutely, this is a special case which is why it is only a warning. Do we need a flag to disable the warning?

Reply:
I used the warnings package so that it is only raised the first time. In additon, I added a suppression flag.